atomic_reactor/plugins/pre_fetch_sources.py

```diff
@@ -229,6 +244,62 @@ def _get_remote_urls_helper(self, koji_build):
 
         return remote_sources_urls, remote_json_map
 
+    def _get_kojifile_source_urls_helper(self, koji_build):
+        """Fetch kojifile source urls from specific build
+
+        :param koji_build: dict, koji build
+        :return: list, dicts with URL pointing to kojifile sources
+        """
+
+        self.log.debug('get kojifile_source_urls: %s', koji_build['build_id'])
+        images = self.session.listArchives(koji_build['build_id'], type='image')
+
+        self.log.debug('images: %s', images)
+
+        sources = []
+        pnc_map = get_pnc(self.workflow)
+        # using urljoin here causes the API path in the base_api_url to be removed
+        api_request_url = pnc_map['base_api_url'] + '/' + pnc_map['get_scm_archive_path']
+        req_session = get_retrying_requests_session()
+
+        kojifile_build_ids = {kojifile['build_id'] for image in images
+                              for kojifile in self.session.listArchives(imageID=image['id'],
+                                                                        type='maven')}
+
+        for build_id in kojifile_build_ids:
+            source_build = self.session.getBuild(build_id, strict=True)
+            if source_build['owner_name'] == PNC_SYSTEM_USER:
+                pnc_build_id = source_build['extra']['external_build_id']
+                url, dest_filename = pnc.get_scm_archive_from_build_id(
+                                                                    api_request_url=api_request_url,
+                                                                    build_id=pnc_build_id,
+                                                                    session=req_session)
+                source = {'url': url,
+                          'subdir': source_build['nvr'],
+                          'dest': '__'.join([source_build['nvr'], dest_filename])}
```

Comment:
why are you prepending nvr to filenames? when we have already nvr directory?

Reply:
When we pass the directories to the BSI, the directory name is not stored. It becomes a tar named exta-src#.tar when # is the layer index. To distinguish sources for given nvrs we have to append it to the filename itself.

Reply:
ah right ok